### PR TITLE
BZ 1289427 Configure and View the Statistics Page for the HAProxy Router

### PR DESCRIPTION
**Pull request by mburke5678**
https://bugzilla.redhat.com/show_bug.cgi?id=1289427

### CODE REVIEW

**Comment by mburke5678** [comment]
@jcantrill This change is from a tech support request to add the steps needed to access the HAProxy router stats page. Is this in your team's purview? Can you PTAL? I could not get the procedure to work, but my environment might not be configured as a user's would be.

**Comment by jcantrill** [comment]
I dont see anything in here that is specific to logging for which i can intelligently comment upon

File: admin_guide/router.adoc
Comment (zhaozhanqi) [inline admin_guide/router.adoc]: path /haproxy_stats has been deprecated from 3.6
cc@knobunc

Comment (mburke5678) [reply admin_guide/router.adoc]: @knobunc ^^

**Comment by mburke5678** [comment]
@knobunc

**Comment by mburke5678** [comment]
@thoraxe You worked on a training document on this issue [05-Services-Routing-Complete-Example.md](https://github.com/openshift/training/blob/master/05-Services-Routing-Complete-Example.md#viewing-router-stats). I cannot make the steps work and and getting timeout errors trying to launch the stats page (it might be due to my using QuickLab systems for my master; not sure what ability I have to expose ports). 
Can you help with this issue?

File: admin_guide/router.adoc
Comment (knobunc) [reply admin_guide/router.adoc]: We should encourage people to use the prometheus stats.  @pecameron we need to make sure that we re-sync when we work out what the oadm command ends up looking like.

Comment (knobunc) [reply admin_guide/router.adoc]: BTW the block at line 115 should be put in its own section.  How about "Retrieving HAProxy Stats Locally"

That's the usual way that I grab the stats... it always works.

**Review by knobunc — CHANGES_REQUESTED** [review]
We are somewhat blocked on this until we work out how we want the `oadm router` command to work now.

@pecameron has PR https://github.com/openshift/origin/pull/16621 open... and the solution is under discussion.  But it may require some changes in this area.

**Comment by mburke5678** [comment]
@knobunc Should we hold off documenting how to expose stats and launch the stats page until you find a solution?

**Comment by pecameron** [comment]
@knobunc @mburke5678 I think this should wait a while. Things seem to be in flux.
By default we setup prometheus statistics through software integrated into the router. We previously used a sidecar. To turn off stats you need 'oc adm router --type= --stats-port=0' the --type must be set to something besides haproxy-router. --expose-metrics and --metrics-image are deprecated. There is more....

**Comment by pecameron** [comment]
@vikram-redhat @mburke5678 @knobunc We seem to be going toward prometheus format metrics, at least in networking. The docs talk about a lot of different metrics and is very confusing. Are we looking to have a high level presentation of OCP metrics that the metrics discussion in the router and the new prometheus metrics in SDN can reference? Are we planning to remove the deprecated discussions? How do we want customers to think about cluster metrics, how to capture them, and how to interpret what they see?

**Comment by thoraxe** [comment]
@mburke5678 I wouldn't expect anything from that old training content to still be considered accurate. I have no idea how or if the router stats port can still be accessed. I'm sorry I'm no help.

**Comment by pecameron** [comment]
@mburke5678 Hold off on the type= stats-port=0 changes. Clayton and I are still changing code. Its looking like the original will work...

**Comment by pecameron** [comment]
@mburke5678 origin 16621 is approved and in the merge queue. No changes are need to the docs. When stats-port=0 metrics are suppressed.

**Comment by mburke5678** [comment]
@knobunc @pecameron @pecameron @sferich888 What about the original request to show how to launch the HAProxy stats page? Do we still want to persue that?

**Comment by mburke5678** [comment]
@knobunc @pecameron @pecameron @sferich888
I verified (in 3.5) that the procedure works to display stats in the browser, as was the original request in this PR. Do we want to publish this change as we work through Phil's work on documenting metrics?

**Comment by pecameron** [comment]
@knobunc Is there something you want to change here?

**Comment by pecameron** [comment]
@knobunc Is there more to do here?
oadm router hap --service-account=router --stats-port=0
works as expected. (after bug fix)

**Comment by pecameron** [comment]
@knobunc its been a while. Is there anything else to change?

**Review by knobunc — APPROVED** [review]
LGTM.  Thanks.  Sorry for the delay.

**Comment by mburke5678** [comment]
[rev_history]
|xref:../ admin_guide/router.adoc#admin-guide-router[Monitoring and Debugging Routers]
|Added new section on xref:../ admin_guide/router.adoc#admin-guide-router-view-stats[Viewing Statistics].
%

**Comment by mburke5678** [comment]
@openshift/team-documentation PTAL

**Comment by mburke5678** [comment]
Note to self: After merge to master, do follow up, 3.6+ to add `oc env dc router ROUTER_METRICS_TYPE-` and ` http://admin:$STATS_PASSWORD@ip:1936/metrics`
https://bugzilla.redhat.com/show_bug.cgi?id=1467257
@zhaozhanqi @knobunc This is correct for 3.6+ ??

**Comment by mburke5678** [comment]
Follow up https://github.com/openshift/openshift-docs/pull/6396

**Comment by mburke5678** [comment]
@jcantrill @pecameron @zhaozhanqi @knobunc 
Do we want to document this feature for 3.7? If so, I assume the changes from #6396 I made for 3.6 would apply?

cc. @bfallonf

**Comment by mburke5678** [comment]
@jcantrill @pecameron @zhaozhanqi @knobunc
I never got an answer to this, so the changes didn't make it to 3.7+ 
>>Do we want to document this feature for 3.7? If so, I assume the changes from #6396 I made for 3.6 would apply?
cc. @kalexand-rh

**Comment by mburke5678** [comment]
@jcantrill @pecameron @zhaozhanqi @knobunc 
Do we want to document this feature for 3.7? If so, I assume the changes from #6396 I made for 3.6 would apply?

**Comment by knobunc** [comment]
Those docs are correct... but you are probably better just hooking Prometheus up to the router stats and then querying Prometheus for the right data.  I'm not sure if there are good docs for any of that that we can crib.

**Pull request merged**
